client/client.go

```diff
@@ -66,27 +66,33 @@ func (c Client) withProject(project string) *Client {
 	return &c
 }
 
-func (c *Client) configureEnabledServices() error {
+func (c *Client) configureEnabledServices() diag.Diagnostics {
 	var esLock sync.Mutex
+	var diags diag.Diagnostics
 	g, ctx := errgroup.WithContext(context.Background())
 	maxGoroutines := limit.GetMaxGoRoutines()
 	goroutinesSem := semaphore.NewWeighted(helpers.Uint64ToInt64(maxGoroutines))
 	for _, p := range c.projects {
 		project := p
 		if err := goroutinesSem.Acquire(ctx, 1); err != nil {
-			return diag.WrapError(err)
+			return diags.Add(diag.WrapError(err))
 		}
 		g.Go(func() error {
 			defer goroutinesSem.Release(1)
 			cl := c.withProject(project)
 			svc, err := cl.fetchEnabledServices(ctx)
 			esLock.Lock()
-			c.EnabledServices[project] = svc
+			if err != nil {
+				c.logger.Warn("failed to fetch enabled services", "project_id", project, "error", err)
+				diags = diags.Add(diag.NewBaseError(fmt.Errorf("failed to fetch enabled services, project_id=%s, error=%w", project, err), diag.USER, diag.WithSeverity(diag.WARNING)))
```

Comment:
nit: would maybe use with summary / details for internal + projrcy id info

Reply:
What is difference between the message in the error vs `summary` vs `details` ? what should be shown in each field and how/where it is displayed to the user?

Reply:
Usually summary is error unless otherwise specified, its a more human readable error, it should be used if for example we are wrapping an existing error.
Details add more information on the error usually user specific information  and/or remediation information